pulpcore/app/wsgi.py

```diff
@@ -9,9 +9,12 @@
 
 import os
 
-from django.core.wsgi import get_wsgi_application  # noqa
+from django.core.wsgi import get_wsgi_application
+from opentelemetry.instrumentation.wsgi import OpenTelemetryMiddleware
+
 
 os.environ.setdefault("DJANGO_SETTINGS_MODULE", "pulpcore.app.settings")
 
 
 application = get_wsgi_application()
+application = OpenTelemetryMiddleware(application)
```

Comment:
```suggestion
try:
    from opentelemetry.instrumentation.wsgi import OpenTelemetryMiddleware
    OTEL_AVAILABLE = True
except ImportError:
    OTEL_AVAILABLE = False


os.environ.setdefault("DJANGO_SETTINGS_MODULE", "pulpcore.app.settings")


application = get_wsgi_application()
if OTEL_AVAILABLE:
    application = OpenTelemetryMiddleware(application)
```

It's not hard to make it optional. And then we don't need to depend on a prerelease of a dependency.

Reply:
Yes but that makes it all more complicated. I want to keep this very simple. Pulp has way too many options.

Reply:
I disagree on all points:
1. That is no excuse for the change provided here. The other more intricate pieces can just about wait until the library we intend to use is released by their maintainers. And quite frankly this piece should too.
2. We are introducing code that gets executed on every single request to the api, and the original authors are not confident to use it in production. "It works" is only about the happy path.
3. It is not the only way that the software is delivered. Pulp is packaged too. Also we should not provide the single most pushed deployment method with unfinished software.